ITISFoundation/osparc-simcore
# Guided mode/partial pipelines (backend only)

## What do these changes do?


allow running partial pipelines as required by the guided mode

**Changes:**
Director-v2:
- POST /computations/{project_uuid}:start API now allow to path a ```subgraph``` field which is a list of nodes to be run
  - a minimal dependency graph is computed by checking if the dependent nodes have an output. in case they don't they will be run as well (NOTE: currently outdated outputs are not checked so it will not automatically re-run if some input number is changed)
  - a selected node is automatically set as ```dirty``` and will re-trigger any node that depend on it
- GET /computations/{project_uuid} now returns the state of the entire or sub-pipeline depending how it was started


Webserver
- /computations/{project_uuid}:start API now allow to path a ```subgraph``` field that is forwarded to director-v2


NOTE: The frontend will come in a next PR by @odeimaiz 

## Related issue number




## How to test
- not really testable without hacking in the frontend...


but the idea is:
now the UI just need to say:

- hey I want to run that node
- the backend will run this node and before that all the dependent ones if needed (with some restrictions, currently the only check is to see if there is an output or not)

the UI can also say:
- I want to run this node, this one and this other one
- the backend will automatically run them with the dependent ones if needed



## Checklist
